src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Msmq/MsmqCommon.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Text;
+using Datadog.Trace.ClrProfiler.Integrations;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Msmq
+{
+    internal class MsmqCommon
+    {
+        internal const string IntegrationName = nameof(IntegrationIds.Msmq);
+        internal static readonly IntegrationInfo IntegrationId = IntegrationRegistry.GetIntegrationInfo(IntegrationName);
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(MsmqCommon));
+
+        internal static Scope CreateScope(Tracer tracer, string command, string spanKind, IMessageQueue messageQueue, bool? messagePartofTransaction = null)
```

Comment:
If we use `IMessageQueue` as parameter then the proxy struct will be boxed. You can use the generic constraint as in the integration.